### PR TITLE
{perf}[gompi/2021a] OSU-Micro-Benchmarks v5.7.1, NCCL v2.10.3, magma v2.6.1 with CUDA 11.3.1

### PR DESCRIPTION
**Pull request by Micket**
(created using `eb --new-pr`)

This is an example of what we could choose to do for CUDA stuff, ref.
https://github.com/easybuilders/easybuild-easyconfigs/issues/12484

Shouldn't be merged until we have agreed on what to do.

### CODE REVIEW

File: easybuild/easyconfigs/o/OSU-Micro-Benchmarks/OSU-Micro-Benchmarks-5.7.1-gompi-2021a-CUDA-11.3.1.eb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+source_urls = ['https://mvapich.cse.ohio-state.edu/download/mvapich/']
+sources = [SOURCELOWER_TGZ]
+
+dependencies = [
+    ('CUDAcore', '11.3.1', '', True),
+    ('UCX-CUDA', '1.10.0', versionsuffix),
+]
+
+configopts = 'CC="$MPICC" CXX="$MPICC"'
+configopts += ' --enable-cuda --with-cuda=$EBROOTCUDA'
```
Comment (akesandgren) [inline easybuild/easyconfigs/o/OSU-Micro-Benchmarks/OSU-Micro-Benchmarks-5.7.1-gompi-2021a-CUDA-11.3.1.eb:21]: Shouldn't that be EBROOTCUDACORE?

Comment (Micket) [reply easybuild/easyconfigs/o/OSU-Micro-Benchmarks/OSU-Micro-Benchmarks-5.7.1-gompi-2021a-CUDA-11.3.1.eb]: yes (it seems osu micro benchmarks managed to find all the cuda stuff anyway)

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/c13daa7ce0694b4bd4faaa81f61e2071 for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 2 out of 3 (3 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/df99c608bd99fba60d65ffc1dae9cde5 for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 2 out of 3 (3 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/67793aa435f01a85fe5adeb0bd045e68 for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/b002e1a2e6799b1fbaa36b73effe1fa3 for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/1a5cbd6024ee2406089c6170a0edce05 for a full test report.

**Comment by Micket** [comment]
Not a lot of input on this yet. As far as I know, only vaguely positive responses favoring this approach, so I suggest to open this up for review.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (3 easyconfigs in total)
node3301.joltik.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz (cascadelake), Python 3.6.8
See https://gist.github.com/f2c6e48496ea200f4cae1372c8816f01 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
jrlogin11.jureca - Linux centos linux 8.3.2011, x86_64, AMD EPYC 7742 64-Core Processor, Python 3.6.8
See https://gist.github.com/febe44f01dbf166b91b1ae214c08452a for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
jwvis00.juwels - Linux centos linux 8.3.2011, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 3.6.8
See https://gist.github.com/e975e939bd2fbb1f58b1b0254696dacf for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso
EB_ARGS=" --buildpath $HOME/build"

**Comment by boegel** [comment]
Going in, thanks @Micket!

**Pull request merged**